packages/scandipwa/src/store/MyAccount/MyAccount.dispatcher.js

```diff
@@ -203,6 +203,52 @@ export class MyAccountDispatcher {
         );
     }
 
+    /**
+     * Resend confirmation email
+     * @param {{email: String}} [options={}]
+     * @memberof MyAccountDispatcher
+     */
+    resendConfirmation(options = {}, dispatch) {
+        const mutation = MyAccountQuery.getResendConfirmationMutation(options);
+
+        return fetchMutation(mutation).then(
+            /** @namespace Store/MyAccount/Dispatcher/MyAccountDispatcher/resendConfirmation/fetchMutation/then */
+            (data) => {
+                const { resendConfirmationEmail: { status = '' } } = data;
+
+                switch (status) {
+                case 'account_confirmation_not_required':
+                    dispatch(showNotification('success', __('This email does not require confirmation.')));
+                    history.push('/customer/account/login');
+
+                    return false;
+                case 'confirmation_sent':
+                    dispatch(showNotification('success', __('Please check your email for confirmation key.')));
+
+                    return true;
+                case 'wrong_email':
+                    const { email = '' } = options;
+
+                    dispatch(showNotification('error', __('Wrong email! Please, try again!')));
+                    history.push(`/customer/account/confirmation/?email=${ email }`);
+
+                    return 'error';
```

Comment:
I believe there is an architectural issue.
The function returns either true, false, or string value "error".
When you refactor this to use async / await, please make it throw an error in case of error.
Then handle the error in the caller function.